centaur/src/main/scala/centaur/test/FilesChecker.scala

```diff
@@ -0,0 +1,95 @@
+package centaur.test
+
+import com.amazonaws.services.s3.AmazonS3
+import com.google.cloud.storage.Storage.BlobListOption
+import com.google.cloud.storage.{Blob, Storage}
+
+import scala.collection.JavaConverters._
+import scala.language.implicitConversions
+
+sealed trait FilesChecker {
+  def countObjectsAtPath: String => Int
+}
+
+case object PipelinesFilesChecker extends FilesChecker {
+
+  import GCS._
+
+  private lazy val storage = Operations.storage
+
+  def countObjectsAtPath: String => Int =
+    storage.parsePath andThen storage.countObjectsAtPath
+}
+
+case object LocalFilesChecker extends FilesChecker {
+  def countObjectsAtPath: String => Int = { s =>
+    val d = new java.io.File(s)
+    if (d.exists && d.isDirectory)
+      d.listFiles.length
+    else if (d.exists && d.isFile)
+      1
+    else
+      0
+  }
+}
+
+case object AWSFilesChecker extends FilesChecker {
+  import AwsS3._
+
+  private lazy val s3Client = Operations.buildAmazonS3Client
+
+  override def countObjectsAtPath: String => Int =
+    s3Client.parsePath andThen s3Client.countObjectsAtPath
+}
+
+case class GCSPath(bucket: String, directory: String)
+
+case class AwsS3Path(bucket: String, directory: String)
+
+object GCS {
+
+  implicit def gcsOps(s: Storage): GCSOps = GCSOps(s)
+}
+
+object AwsS3 {
+  implicit def awsOps(amazonS3: AmazonS3): AwsS3Ops = AwsS3Ops(amazonS3)
+}
+
+case class AwsS3Ops(amazonS3: AmazonS3) {
+
+  def parsePath: String => AwsS3Path = { fullPath =>
+    val bucketAndDashes = fullPath.drop(5).split('/')
```

Comment:
You can use another `split` method, which is `split(regex, limit)`
In other words, `split("/", 2)` will do the job.